### PR TITLE
Rename setup_classes & include creation_date in ExperimentInfo

### PR DESCRIPTION
**Pull request by lewisblake**
## Change Summary

- Renamed `setupclasses.py` to `setup_classes.py` to follow Python module naming standard of splitting words with underscores
- Added a `creation_date` attribute to `ExperimentInfo`. Defaults to `datetime.today().strftime("%Y-%m-%d")` but can also be provided if a pyaeroval user wants to modified it.

## Related issue number

Close https://github.com/metno/AeroToolsIssues/issues/75

## Checklist

* [x] Start with a draft-PR
* [x] The PR title is a good summary of the changes
* [x] PR is set to AeroTools and a tentative milestone
* [ ] Documentation reflects the changes where applicable
* [x] Tests for the changes exist where applicable
* [x] Tests pass locally
* [x] Tests pass on CI
* [x] At least 1 reviewer is selected
* [x] Make PR ready to review

### CODE REVIEW

File: pyaerocom/aeroval/setup_classes.py
Comment (heikoklein) [inline pyaerocom/aeroval/setup_classes.py]: Please use a json datestamp like `2012-04-23T18:25:43.511Z` please, let the designer decide how it looks on the web.

**Review by heikoklein — CHANGES_REQUESTED** [review]
The official interface for using `EvalSetup` is:
```from pyaerocom.aeroval import EvalSetup```
Therefore, renaming setupclasses to setup_classes is fine here. It would be good if the test-cases could use the official interface rather than the internal access-path.

The documentation for EvalSetup needs to been fixed: https://github.com/metno/pyaerocom/blob/be7aeeff06f5ea49b38c3a717567c4a516beacf2/docs/api-aeroval.rst?plain=1#L14

Can creation_date have time-resolution e.g. https://stackoverflow.com/questions/10286204/what-is-the-right-json-date-format

File: pyaerocom/aeroval/setup_classes.py
Comment (lewisblake) [reply pyaerocom/aeroval/setup_classes.py]: The updates will serialize to a non-timezone aware string like this, if you're okay with it. 
`{'creation_date': '2024-09-19 15:46:36.036175'}`

`datetime.now()` returns a datetime in your local time zone, but it is not timezone-aware.

Comment (heikoklein) [reply pyaerocom/aeroval/setup_classes.py]:
```
creation_date: str = f"{datetime.datetime.now(datetime.timezone.utc):%Y-%m-%dT%H:%M:%S.%fZ"
```

File: tests/aeroval/test_aeroval_HIGHLEV.py
Comment (heikoklein) [inline tests/aeroval/test_aeroval_HIGHLEV.py]: Use the official way for importing EvalSetup:

`from pyaerocom.aeroval import EvalSetup`

File: tests/aeroval/test_experiment_output.py
Comment (heikoklein) [inline tests/aeroval/test_experiment_output.py]: import without .setup_classes

File: tests/aeroval/test_experiment_processor.py
Comment (heikoklein) [inline tests/aeroval/test_experiment_processor.py]: drop setup_classes

File: tests/aeroval/test_helpers.py
Comment (heikoklein) [inline tests/aeroval/test_helpers.py]: drop setup_classes

File: tests/aeroval/test_modelmaps_engine.py
Comment (heikoklein) [inline tests/aeroval/test_modelmaps_engine.py]: drop setup_classes

File: tests/aeroval/test_setup_classes.py
Comment (heikoklein) [inline tests/aeroval/test_setup_classes.py:7]: :-)

**Review by heikoklein — CHANGES_REQUESTED** [review]
Please use the json-datetime format and utc.
Please use the official way to import EvalSetup where possible.

File: tests/aeroval/test_setup_classes.py
Comment (lewisblake) [reply tests/aeroval/test_setup_classes.py:7]: Whoops, sorry, totally messed that up last time. Should be fixed now!

File: pyaerocom/aeroval/setup_classes.py
Comment (lewisblake) [reply pyaerocom/aeroval/setup_classes.py]: Changed

File: tests/aeroval/test_experiment_output.py
Comment (heikoklein) [inline tests/aeroval/test_experiment_output.py:116]: To avoid hardcoding class-pathes, it could be written as:
```
f"need instance of {EvalSetup}"
```

**Review by heikoklein — APPROVED** [review]
Thanks.

I just got a small idea on how to hide the `setup_classes` in user/test-space even more, not crucial, so I approve the PR and leave it to you if you want to change or just merge.

File: tests/aeroval/test_experiment_output.py
Comment (lewisblake) [reply tests/aeroval/test_experiment_output.py:116]: I think this `ValueError.value` comes from https://github.com/metno/pyaerocom/blob/be7aeeff06f5ea49b38c3a717567c4a516beacf2/pyaerocom/_lowlevel_helpers.py#L105

so it would be a little tricky to change just in this case.

Comment (heikoklein) [reply tests/aeroval/test_experiment_output.py:116]: I just meant replace 
```python
assert str(e.value) == "need instance of <class 'pyaerocom.aeroval.setupclasses.EvalSetup'>"
```
with
```python
assert str(e.value) == f"need instance of {EvalSetup}"
```
This would not need any changes to _lowlevel_helpers.py. The string expansion of `{self._type}` and `{class}` is the the same.

**Pull request merged**
